inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -758,24 +768,57 @@ void MKLDNNGraph::PullOutputData(BlobMap &out) {
             ext_blob->allocate();
         }
 
-        if (ext_blob->byteSize() != intr_blob.GetSize())
-            THROW_IE_EXCEPTION << "Output blob size is not equal network output size ("
-                               << ext_blob->size() << "!=" << intr_blob.GetSize()/sizeof(float) << ").";
-
         void *ext_blob_ptr = ext_blob->buffer();
         void *intr_blob_ptr = intr_blob.GetData();
 
-        // That is the same memory. No need to copy
-        if (ext_blob_ptr == intr_blob_ptr) continue;
-
         int MB = intr_blob.GetDims()[0];
         int MB_to_process = node->batchToProcess();
         // TODO: Should we support InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_LIMIT???
         if (config.batchLimit)
             MB_to_process = std::min<int>(config.batchLimit, MB_to_process);
         size_t size_to_copy = intr_blob.GetSize() * MB_to_process / MB;
 
-        cpu_memcpy_s(ext_blob_ptr, ext_blob->byteSize(), intr_blob_ptr, size_to_copy);
+        auto srcPrec = MKLDNNMemory::convertToIePrec(intr_blob.GetDataType());
+        auto dstPrec = ext_blob->getTensorDesc().getPrecision();
+        if (dstPrec == srcPrec || (dstPrec == Precision::BOOL && srcPrec == Precision::U8)) {
```

Comment:
What do you think about handling of `dstPrec == srcPrec` case inside convert itself? So if typename src == typename dst you will be able to call cpu_memcpy_s. This will allow you not to check such custom cases as `dstPrec == Precision::BOOL && srcPrec == Precision::U8`

Reply:
If I understood your idea correctly, then it will look like this: we create list precision which we directly support in plugin (including ```BOOL```), if ```dstPrec``` is included in this list we call instance of ```convert```, else process as now.
I believe we don't need this. It is increase binary size and we don`t have many such specific cases like BOOL and U8

Reply:
We can consider the option to extract ```Convert Layer``` execute function and use it if we need this, it's not increase binary size and we got more simple copy function